src/prefect/core/task.py

```diff
@@ -755,7 +755,6 @@ def map(
                     )
                 )
         task_args = task_args.copy() if task_args else {}
-        task_args.setdefault("nout", None)
```

Comment:
For my own understanding, why did this need to be removed?

Reply:
If that line is kept, `__iter__` therefore task unpacking cannot be called on mapped tasks. `__iter__` throws a value error for tasks without nout specified. 
This used to be correct since we didn't allow task unpacking. However this PR aims to change that, i.e. it aims to bring nout support for mapped tasks.